src/NuGet.Clients/NuGet.CommandLine/Commands/AddCommand.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.CommandLine
+{
+    [Command(typeof(NuGetCommand), "add", "AddCommandDescription;DefaultConfigDescription",
+        MinArgs = 1, MaxArgs = 1, UsageDescriptionResourceName = "AddCommandUsageDescription",
+        UsageSummaryResourceName = "AddCommandUsageSummary", UsageExampleResourceName = "AddCommandUsageExamples")]
+    public class AddCommand : Command
+    {
+        [Option(typeof(NuGetCommand), "AddCommandSourceDescription", AltName = "src")]
+        public string Source { get; set; }
+
+        [Option(typeof(NuGetCommand), "ExpandDescription")]
+        public bool Expand { get; set; }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            // Arguments[0] will not be null at this point.
+            // Because, this command has MinArgs set to 1.
+            var packagePath = Arguments[0];
+            OfflineFeedUtility.ValidatePath(packagePath);
```

Comment:
As we discussed let change the Validate method to do all the needed validations and also change the method name to ThrowIfInvalid~